graph_snyk_cve_sync.py

```diff
@@ -0,0 +1,358 @@
+#!/usr/bin/env python3
+# Copyright © 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Author: Yusuf Zainee <yzainee@redhat.com>
+#
+
+"""Script which synchronizes snyk CVEs from S3 to graph."""
+
+from datetime import datetime, timedelta
+import boto3
+import json
+import os
+from f8a_utils.versions import get_versions_for_ep
+from semantic_version import Version, NpmSpec
+from unified_range import api
+from f8a_version_comparator.comparable_version import ComparableVersion
+import re
+import logging
+from requests_futures.sessions import FuturesSession
+import time
+
+CVE_BUCKET = os.environ.get("REPORT_BUCKET_NAME", '')
+AWS_KEY = os.environ.get("AWS_S3_ACCESS_KEY_ID_REPORT_BUCKET", '')
+AWS_SECRET = os.environ.get("AWS_S3_SECRET_ACCESS_KEY_REPORT_BUCKET", '')
+AWS_REGION = os.environ.get("AWS_S3_REGION", "us-east-1")
+DRY_RUN = os.environ.get('DRY_RUN', 'false').lower() in ('1', 'yes', 'true')
+HOST = os.environ.get('BAYESIAN_DATA_IMPORTER_SERVICE_HOST', 'bayesian-data-importer')
+PORT = os.environ.get('BAYESIAN_DATA_IMPORTER_SERVICE_PORT', '9192')
+
+s3_resource = boto3.resource('s3', aws_access_key_id=AWS_KEY,
+                             aws_secret_access_key=AWS_SECRET,
+                             region_name=AWS_REGION)
+
+CVE_DATA = {}
```

Comment:
IMHO, we should use the Object-Oriented approach here.
Benefits of using OOP here:
- Can be helpful in inheritance. (In future: For any new vendor integration, `get_version_rules` and others can be inherited and reused)
- `CVE_DATA` is globally declared, Classes could help in definitive namespacing of variables and methods.
- Helps in readability too.

Reply:
I think we should encapsulate all new functions in a class.